### DISCUSSION TITLE
Considering supporting OpenSearch Dashboards ?

### DISCUSSION THREAD

**Discussion by jbgachot** [fbaligand/kibana-enhanced-table#211] [Ideas]
Hello !

First, thanks a lot with the enhanced table plugin ! Great work ! ❤️ 

Then I wanted to start here a discussion as I couldn't find any related subject in this repository yet :

**With the recent release of [OpenSearch Dashboards 1.0.0-beta1](https://github.com/opensearch-project/OpenSearch-Dashboards) based on kibana 7.10.2, would you consider supporting it ?**

This will imply some work on changing names here and there, but overall it should be relatively easy to port it !

🤟

**Comment by stef29Brest** [comment]
bonjour, (en Français désolé) , je suis aussi très intéressé pas le support sur kibana 7.10.2, j'ai passé quelques soirées à tenter l'ajout du module à mon kibana sur opendistro.
hello,  I am also very interested in the support on kibana 7.10.2, I spent a few evenings trying to add this module to my kibana on opendistro_

**Reply by fbaligand** [reply]
Hi, kibana 7.10.2 is already supported. 
you can find enhanced table package for kibana 7.10.2 here:
https://github.com/fbaligand/kibana-enhanced-table/releases/tag/v1.11.0

**Comment by fbaligand** [comment]
> Hello !
> 
> First, thanks a lot with the enhanced table plugin ! Great work ! ❤️
> 
> Then I wanted to start here a discussion as I couldn't find any related subject in this repository yet :
> 
> **With the recent release of [OpenSearch Dashboards 1.0.0-beta1](https://github.com/opensearch-project/OpenSearch-Dashboards) based on kibana 7.10.2, would you consider supporting it ?**
> 
> This will imply some work on changing names here and there, but overall it should be relatively easy to port it !
> 
> 🤟

Hi @jbgachot,
First, sorry for my late reply, I totally missed your message until now. 
So to answer you, little changes can be done in plugin code to support open dashboards, but no breaking change for plugin users. 
by the way, feel free to open a PR for that. 

Fabien

**Reply by fbaligand** [reply]
But are there differences between OpenSearch Dashboards and Kibana 7.10.2 ?

**Reply by jbgachot** [reply]
Hello ! Also sorry for the late reply 😉 

Well there should be no difference (yet) in functionality, but it would mean some modifications in the code :
Something like "grep"ing kibana and changing to opensearch-dashboards. Because what opensearch did is basically refactor the whole code to get rid of the brand "kibana" from version 7.10.2.

But I imagine it would mean to maintain two different code base because of the functions / files / directories refactor.

And eventually opensearch will have there own bug fixes and features.

**Reply by fbaligand** [reply]
Hi,

Honestly, I'm not very comfortable with the idea to create a "branch" for OpenSearch Dashboards.
I mean: I already maintain the plugin for every Kibana version and it is a big work, not simple.
But giving what you say, it seems that creating a branch for opensearch dashboards would be the only way.

Concerning fixes and features, most of the time, I create the features  on master branch, and then backport the commits to the different branches (that are for other Kibana versions). So if there is a branch for opensearch dashboards, it would be the same, and so no specific feature for Kibana or OpenDashboards just little changes in the way to implement.

**Comment by jbgachot** [comment]
Hello Fabien.

Long time no see. I finally wrote the commit [here](https://github.com/jbgachot/kibana-enhanced-table/commit/2c1f74c7bd3bf01300154d3c01a49bdf828206c5) 🎉 

I used v1.11.0 since Opensearch Dashboards is based on Kibana 7.10.2

Not much to say about this commit except the rebranding made by the Opensearch Project. I have been able to test all the 'enhanced tables' I have in my historical Kibana instance and everything works as expected for now on OpenSearch Dashboards 1.2.0

Let me know what you think 😃

**Comment by fbaligand** [comment]
Hi @jbgachot,

First, sorry to answer you so late.
Then, thank you for your great work on opensearch dashboards compatibility! That's awesome!
By the way, happy to see that you tested it, and all works as expected!

Now, I have some remarks before it can be merged on enhanced-table repository:
- first, you maybe don't notice, but I created a "opensearch-dashboards" branch (based on 7.9 branch that is compatible with Kibana 7.9 to 7.11). This morning, I added a commit to change 'Development' section in README, specifically for Opensearch-Dashboards build.
-  can you rebase your commit on 'opensearch-dashboards' branch? If you need some help, tell me.
- then, I see that you replace all `kibana` or `kbn` terms by `opensearchdashboards` or `osd`. I fully understand that globally Opensearch-Dashboards aims to remove all kibana trademarks in the code. Specifically on enhanced-table, this is necessary for package.json, kibana.json and all imports in code from kibana core. That said, I'm a bit annoyed to rename every 'kibana' or 'kbn' inside 'enhanced-table' code itself. I explain. Regularly, I add bug fixes or enhancements to enhanced-table (I'm currently working on a new release). First, I do the work on master branch, and then I back-port the changes on other branches (using `git cherry-pick`). If opensearch-dashboards branch has too much differences, it is a nightmare to merge the changes, and so, I want to avoid this kind of hard work. So, is it possible to not rename internal variables/folders/objects in enhanced-table?
- finally, it would be great that you open a pull request on [opensearch-dashboards](https://github.com/fbaligand/kibana-enhanced-table/tree/opensearch-dashboards) branch!

**Comment by jbgachot** [comment]
Hello Fabien ! No worries.

First of all, i did saw the opensearch-dashboards branch but I though that v1.11.0 was the best choice for opensearch since it was fully 7.10.x version. But i could rework on opensearch-dashboards branch if you prefer.

Then I got you on the "to minimize the modification to strictly minimum" by keeping variables that don't need modifications, I can make it simpler indeed.

To summarize I will work on that when I come back from my vacations 🎿  😄 

I'm however a bit worried of when OSD will release new features that will diverge from KBN. Will it not be too much trouble then ?

See you soon

**Comment by fbaligand** [comment]
Hi,

First, I wish you a great holiday! :)

Then, 7.9 branch (and so currently opensearch-dashboards branch) is compatible from Kibana 7.9 to 7.11. I tested it few days ago with Kibana 7.9. Since v1.11.0, there are few enhancements on the branch. You can see commits history.
So yes, I would prefer that your rebase your commit on opensearch-dashboards branch.
Thus, the aim is to merge your PR onto this branch :)

Finally, it is very difficult to predict how OSD will diverge from Kibana...
That said, Kibana itself has changed a lot through the different versions. Sometimes a minor version brings a big breaking change for enhanced table plugin. And until now, I deal with that. Concretely, I create a branch at each breaking minor version.
So we will do the same way for opensearch-dashboards.

**Comment by fbaligand** [comment]
Hi,

Great news: enhanced-table plugin is now compatible with OpenSearch Dashboards!

You can get enhanced-table package for OSD in last release (v1.13.1) here:
https://github.com/fbaligand/kibana-enhanced-table/releases/tag/v1.13.1

All packages for OSD are suffixed by "_osd-X.Y.Z.zip", where "X.Y.Z" is OSD version.

@jbgachot, many thanks for your commit that helps me a lot to add support for OSD!
